app/controllers/concerns/valid_request.rb

```diff
@@ -34,8 +34,10 @@ def _compute_redirect_to_location(request, options) # :nodoc:
       "#{URL.protocol || request.protocol}#{request.host_with_port}#{options}"
     when Proc
       _compute_redirect_to_location request, instance_eval(&options)
-    else
+    when Hash
       url_for(options)
```

Comment:
Trying to debug this was very difficult. For some reason, I was provided an array to work with, which is not something is made to handle.